### ISSUE TITLE
`vite@3.1.3` breaks `cypress-vite`

### ISSUE THREAD

**Issue by DrJume** [mammadataei/cypress-vite#5]
After updating from `vite@3.1.1` to `3.1.3` I get the following error when running a cypress test: 

```
Error: ENOENT: no such file or directory, stat '/Users/drjume/Library/Application Support/Cypress/cy/production/projects/maler-juergens-1b7c2c9d2b2d995e3f4a8a216021051e/bundles/cypress/e2e/1-request-vacation.cy.ts'
```

**Comment by mammadataei** [comment]
I've updated the `vite` in the example to `3.1.3`, and it works fine. 

https://github.com/mammadataei/cypress-vite/blob/0738fbe8188d2d679982dc2339cc1fbff0c883c7/example/package.json#L31

I'm not sure what's causing this issue, but I've added [debugging](https://github.com/mammadataei/cypress-vite#debugging) at `v1.2.0`; maybe it helps to identify the problem. I can try to fix it if you provide me more information or reproduction if possible.

**Comment by DrJume** [comment]
Thanks for providing a debug output so quickly! I have reproduced the issue and logged it: 

[cypress-vite-debug.log](https://github.com/mammadataei/cypress-vite/files/9657966/cypress-vite-debug.log)

**Comment by mammadataei** [comment]
I've cleared the `projects/` directory and tested with the example again, which still works fine. 
I've also checked the log file, and I think it should be related to the [rollup configuration](https://rollupjs.org/guide/en/#outputinlinedynamicimports).

```
Error: Invalid value for option "output.inlineDynamicImports" - multiple inputs are not supported when "output.inlineDynamicImports" is true.
```

In `vite@3`, the [`output.inlineDynamicImports` is false by default](https://vitejs.dev/guide/migration.html#ssr-changes) and `cypress-vite` is not changing it either. 

If you didn't change it in your `vite.config.ts`, maybe one of vite plugins is changing the configuration. May you check if setting `build.rollupOptions.output.inlineDynamicImports` to `false` can fix the problem?

**Comment by websitevirtuoso** [comment]
@mammadataei Holy Moly.
Thanks for this repo. Finally I can use vite instructions, vite aliases, and everything from vite. 
Thanks for everything.

**Comment by Ragura** [comment]
I'm having the exact same issue, using Vite 3.1.7. I worked with Cypress using the regular Webpack builder first, which created some bundle artifacts that still ran correctly after switching to this package as the builder. But when I manually removed the contents of the projects folder (inside library/cypress/...) they also stopped working.
I have tried setting the `build.rollupOptions.output.inlineDynamicImports` to false without success.

**Comment by mammadataei** [comment]
The only thing I could figure out was that the rollup is unable to transpile modules for some unknown reasons. I need a reproduction to work on this. Can anyone help with that?

@Ragura, it may help if you can send me the debugging output.

**Comment by shwarcu** [comment]
Hi @mammadataei , I am not sure about reproduction yet but in my case, I can't run some tests if I use path aliases in imports
so when I switched from this
```ts
import { HUMAN_LIKE_SCROLL_DURATION, Page } from '../../support/test-configuration';
```
to something like this
```ts
import { HUMAN_LIKE_SCROLL_DURATION, Page } from '@support/test-configuration';
```

I get the error. When I changed all imports back to the previous form, so that no import uses path alias, test worked again.
Then I was changing each import one by one to use path alias, but this time it worked when I made all imports use path alias. Very weird

**Comment by mammadataei** [comment]
@shwarcu Hi, I think, in your case, it can be due to configurations. Did you define path aliases in the vite config?! If not, try to do so and pass the vite config path to `vitePreprocessor()`.

For debugging, I think there is no need for `export`; `DEBUG=cypress-vite` would do.

**Comment by DrJume** [comment]
I very much suspect that https://github.com/vitejs/vite/pull/10116 is causing the error with `cypress-vite`.

**Comment by DrJume** [comment]
To reproduce this error you need to clear the cypress cache for the processed files or use a fresh new project.

**Comment by DrJume** [comment]
I found the cause of the bug:

vitejs/vite#10116 only surfaced the bug. The issue was that other vite plugins (https://github.com/laravel/vite-plugin) interfered with `cypress-vite`, and causing something like multiple rollup inputs.

**Because of this, I am asking myself the question, if it is even suitable to use the plugins from the development build for spec files?**

After I noticed this, I made an empty vite config only for `cypress-vite` and provided is as a parameter to the `viteProcessor()`, but it still used the plugins from the default `vite.config.ts`.

I found out that vite's `build` function has a `configFile` parameter, which when not provided, defaults to the root vite config. (See https://vitejs.dev/guide/api-javascript.html#inlineconfig)
https://github.com/mammadataei/cypress-vite/blob/cc0b5fd20b88cc8962ecbe95edcd1b2c83f52e34/src/index.ts#L73

So it uses the `buildConfig`, but merges it again with the root vite config.

### Workaround
A workaround is specifying `configFile: false` in the vite config given to `viteProcessor()`, because it ends up in `build(buildConfig)`.

### Conclusion
To fix this issue, I switched the config handling to using vite's integrated `configFile` option in my PR #11.

**Comment by DrJume** [comment]
The behaviour of `configFile` is found here:

**Handling option `configFile`:**
https://github.com/vitejs/vite/blob/1a2e7a8d360c6ac25754466143c5a3eb5291dcaa/packages/vite/src/node/config.ts#L392

**Resolving the `configFile`:**
https://github.com/vitejs/vite/blob/1a2e7a8d360c6ac25754466143c5a3eb5291dcaa/packages/vite/src/node/config.ts#L874

**Comment by mammadataei** [comment]
@DrJume Thanks for debugging, detailed explanation, and PR. I merged your PR. I need to do some checks and release it within an hour.

**Comment by mammadataei** [comment]
Should be solved in [v1.2.1](https://github.com/mammadataei/cypress-vite/releases/tag/v1.2.1)
